src/main/java/redis/clients/jedis/BuilderFactory.java

```diff
@@ -97,7 +97,7 @@ public String toString() {
     @SuppressWarnings("unchecked")
     public Map<String, String> build(Object data) {
       final List<byte[]> flatHash = (List<byte[]>) data;
-      final Map<String, String> hash = new HashMap<String, String>();
+      final Map<String, String> hash = new HashMap<String, String>(flatHash.size()/2);
```

Comment:
You might want to add load factor `1` as you already know the target size in advance. Load factor `1` prevents rehashing. Same for the change below.